web/html/src/manager/state/display-highstate.tsx

```diff
@@ -1,99 +1,37 @@
 import * as React from "react";
-import Network from "utils/network";
+import { useState } from "react";
+import HighstateSummary from "./highstate-summary";
 
-function requestHighstate(id) {
-  return Network.get("/rhn/manager/api/states/highstate?sid=" + id).promise;
-}
-
-type MinionHighstateSingleProps = {
-  data: any;
-};
 
-type MinionHighstateSingleState = {
-  highstate?: any;
+function MinionHighstateSingle({ minion }: { minion: { id: number, name: string } }) {
+  return (
+    <div className="row">
+      <h3>{t("State Summary for {0}", minion.name)}</h3>
+      <HighstateSummary minionId={minion.id} />
+    </div>
+  );
 }
 
-class MinionHighstateSingle extends React.Component<MinionHighstateSingleProps, MinionHighstateSingleState> {
-  state: MinionHighstateSingleState = {};
-
-  getHighstate = () => {
-    requestHighstate(this.props.data.id).then(data => {
-      this.setState({ highstate: data });
-    });
-  };
-
-  UNSAFE_componentWillMount() {
-    this.getHighstate();
-  }
-
-  render() {
-    return (
-      <div className="panel panel-default">
-        <div className="panel-heading">
-          <h4>{t("Highstate for {0}", this.props.data.name)}</h4>
-        </div>
-        <div className="panel-body">
-          {this.state.highstate ? <pre>{this.state.highstate}</pre> : <span>Retrieving highstate data...</span>}
-        </div>
-      </div>
-    );
-  }
-}
-
-type MinionHighstateProps = {
-  data: any;
-};
+function MinionHighstate({ minion }: { minion: { id: number, name: string } }) {
+  const [show, setShow] = useState(false);
 
-type MinionHighstateState = {
-  highstate?: any;
-  show: boolean;
-  loading?: boolean;
-};
-
-class MinionHighstate extends React.Component<MinionHighstateProps, MinionHighstateState> {
-  state: MinionHighstateState = { show: false };
-
-  getHighstate = () => {
-    if (this.state.loading) return;
-    this.setState({ loading: true });
-
-    requestHighstate(this.props.data.id).then(data => {
-      this.setState({ highstate: data });
-    });
-  };
-
-  componentDidMount() {
-    if (this.state.show) {
-      this.getHighstate();
-    }
-  }
-
-  expand = () => {
-    this.getHighstate();
-    this.setState({ show: !this.state.show });
-  };
-
-  render() {
-    return (
-      <div className="panel panel-default">
-        <div className="panel-heading" onClick={this.expand} style={{ cursor: "pointer" }}>
-          <span>{this.props.data.name}</span>
+  return (
+    <div className="panel panel-default" style={{ marginBottom: 10 }}>
+      <div className="panel-heading" onClick={() => { setShow(!show) }} style={{ cursor: "pointer" }}>
```

Comment:
This was already here previously, but a small nitpick since we're working on it: if this is something that can be clicked on to change state, it should probably be a `button`.

Reply:
It's technically an accordion and has some specific styling from bootstrap via `panel-heading` so I'm not sure if it's a good idea

Reply:
A better implementation would be to use native accordions from bootstrap all across, but I think it's too much for this PR.

Reply:
Fair enough, just something that caught my eye.